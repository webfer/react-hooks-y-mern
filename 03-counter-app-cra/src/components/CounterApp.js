import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 5 }) => {
  const [counter, setCounter] = useState(value);

  // handleAdd
  const handleAdd = () => {
    setCounter(counter + 1); // This way is valid too
    // setCounter((c) => c + 1); // This way is valid too
  };

  const handleReduce = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleReduce}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
