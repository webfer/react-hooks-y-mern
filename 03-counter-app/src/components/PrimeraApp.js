// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ greet, subtitle }) => {
  // const greet2 = {
  //   name: 'Fernando',
  //   age: 34,
  // };

  return (
    <>
      <h1>{greet}</h1>
      {/* <pre>{JSON.stringify(greet2, null, 3)}</pre> */}
      <p>{subtitle}</p>
    </>
  );
};

PrimeraApp.propTypes = {
  greet: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  subtitle: 'Subtitle aplication by default',
};
export default PrimeraApp;
