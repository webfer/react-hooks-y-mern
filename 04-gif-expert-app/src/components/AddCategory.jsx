import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = ({ target }) => {
    // console.log(event.target.value);
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Search Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
