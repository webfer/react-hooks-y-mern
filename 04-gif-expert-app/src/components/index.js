export * from './AddCategory';
export * from './GifExpertApp';
export * from './GifGrid';
export * from './GifItem';
