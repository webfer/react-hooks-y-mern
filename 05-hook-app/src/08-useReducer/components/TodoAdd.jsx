import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ onNewTodo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: '',
  });

  const onFormSubmit = (event) => {
    event.preventDefault();
    if (description.length <= 1) return;
    const newTodo = {
      id: new Date().getTime(),
      done: false,
      description,
    };
    onNewTodo(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={onFormSubmit}>
      <input
        type="text"
        placeholder="What should be done?"
        name="description"
        className="form-control"
        value={description}
        onChange={onInputChange}
      />
      <button type="submit" className="btn btn-outline-primary mt-3">
        Add
      </button>
    </form>
  );
};
