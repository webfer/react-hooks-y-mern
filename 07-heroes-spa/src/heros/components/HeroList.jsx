import { useMemo } from 'react';
import { getHerosByPublisher } from '../helpers';
import { HeroCard } from './';

export const HeroList = ({ publisher }) => {
  const heros = useMemo(() => getHerosByPublisher(publisher), [publisher]);
  return (
    <>
      <div className="row row-cols-1 row-cols-md-3 g-3">
        {heros.map((hero) => (
          <HeroCard key={hero.id} {...hero} />
        ))}
      </div>
    </>
  );
};
