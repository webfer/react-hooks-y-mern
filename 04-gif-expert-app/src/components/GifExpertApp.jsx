import { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);
  const onAddCategory = (newCategory) => {
    // const item = 'Valorant';
    // setCategories((cat) => [...cat, item]); // Other valit way
    if (categories.includes(newCategory)) return; // Validation for avoid duplicate categories
    setCategories([newCategory, ...categories]);
  };
  return (
    <>
      <h1>Gif Expert App</h1>

      <AddCategory
        // setCategories={setCategories}
        onNewCategory={(event) => onAddCategory(event)}
      />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
