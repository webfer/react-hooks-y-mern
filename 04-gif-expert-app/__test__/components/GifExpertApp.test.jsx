import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../../src/components/GifExpertApp';

describe('Testin on GifExpertApp<GifExpertApp/> component', () => {
  test('should match wit the snapshot', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
});
