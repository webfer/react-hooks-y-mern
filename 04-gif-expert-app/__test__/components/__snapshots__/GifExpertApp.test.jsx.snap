// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testin on GifExpertApp<GifExpertApp/> component should match wit the snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
     
    One Punch
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
